src/app/static/src/app/components/plots/MapLegend.vue

```diff
@@ -30,7 +31,7 @@
                 return [0, 1, 2, 3, 4, 5].map((i) => {
                     const val = Math.round((this.min + (i * step)) * 100)/100;
                     return {
-                        val, style: {background: this.getColor(val/(this.max - this.min))}
+                        val, idx: i, style: {background: getColor(val, this.min, this.max, this.colorFunction as (t: number) => string)}
```

Comment:
This is using the same getColor method as the getters now, which is a bit more defensive about single value ranges etc